File: framework/languages/java/zlink-framework-core/src/main/java/systems/zlink/framework/monitoring/ZLinkListenerStatus.java
package systems.zlink.framework.monitoring;

import java.time.Instant;
import java.util.Objects;

/** The advertised endpoint confirmed for one local listener. */
public record ZLinkListenerStatus(
    ZLinkListenerKind kind,
    String name,
    String endpoint,
    Instant observedAt) {
    public ZLinkListenerStatus {
        Objects.requireNonNull(kind, "kind");
        if (name == null || name.isBlank()) {
            throw new IllegalArgumentException("name is required");
        }
        if (endpoint == null || endpoint.isBlank()) {
            throw new IllegalArgumentException("endpoint is required");
        }
        Objects.requireNonNull(observedAt, "observedAt");
    }
}

File: framework/languages/dotnet/src/Zlink.Framework/Contracts/Configuration/ZLinkListenerStatus.cs
namespace Zlink.Framework.Contracts.Configuration;

public enum ZLinkListenerKind
{
    RouteMesh,
    ClientServer,
    Fanout,
    Stream,
}

public sealed record ZLinkListenerStatus(
    ZLinkListenerKind Kind,
    string Name,
    string Endpoint,
    DateTimeOffset ObservedAt
);
